### ISSUE TITLE
webRes.headers is not iterable

### ISSUE THREAD

**Issue by aarongreenlee** [solidjs/solid#1600]
## tl;dr
webRes.headers is not iterable when attempting to return data from API route. The result is that the connection does not close.

## Background
Today, I setup our first API routes. After a handler executes, an unhandled error is thrown:

```
An unhandled error occured: TypeError: webRes.headers is not iterable
     at handleNodeResponse (file:///Users/aarongreenlee/repos/src/github.com/post-media/post-web/apps/pulitzer/node_modules/solid-start/node/fetch.js:158:38)
     at startHandler (file:///Users/aarongreenlee/repos/src/github.com/post-media/post-web/apps/pulitzer/node_modules/solid-start/dev/server.js:121:7)
     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Example Handler:
```
import { APIEvent } from 'solid-start/api';
import { json } from 'solid-start';

export async function POST(apiEvent: APIEvent) {
  try {
    const body = await new Response(apiEvent.request.body).json()
    const cookies = apiEvent.request.headers.get('Cookie') ?? '';
    const params = {
      ...body,
      ...cookies,
    };
    return '{"success": true}';
  } catch (e) {
    return json(e);
  }
}
```

### Your Example Website or App

post.news

### Steps to Reproduce the Bug or Issue

1. Create an API Handler
2. Make a request to this new handler 
3. Observe the unhandled error
4. Connection from the client is left open

### Expected behavior

1. Create an API Handler
2. Make a request to this new handler
3. Response is returned and the connection is closed

### Screenshots or Videos

![CleanShot 2023-03-07 at 21 00 54@2x](https://user-images.githubusercontent.com/264508/223600059-5bf208bd-1a95-45a6-bb76-ead701d1862d.png)

### Platform

- OS: macOS
- Browser: Any
- Version: Confirmed in 0.2.22 and 0.2.23

### Additional context

Thank you

**Comment by aarongreenlee** [comment]
Moving to solid-start.

**Comment by aarongreenlee** [comment]
See https://github.com/solidjs/solid-start/issues/793
